src/liger_kernel/chunked_loss/fused_linear_distillation.py

```diff
@@ -0,0 +1,278 @@
+            student_logps (torch.Tensor): Avg log probabilities of student inputs. Shape: (batch_size, hidden_size,).
+            teacher_logps (torch.Tensor): Avg log probabilities of teacher inputs. Shape: (batch_size, hidden_size,).
+        """
+        raise NotImplementedError("Distillation loss function must be implemented.")
+
+    @staticmethod
+    def chunk_forward(
+        student_input_chunk,
+        student_weight,
+        teacher_input_chunk,
+        teacher_weight,
+        target_chunk,
+        student_bias=None,
+        teacher_bias=None,
+        ignore_index=-100,
+        compute_ce_loss=True,
+    ):
+        # Student
+        student_logits_chunk = student_input_chunk @ student_weight.t()
+        if student_bias is not None:
+            student_logits_chunk = student_logits_chunk + student_bias
+        student_log_probs_chunk = F.log_softmax(student_logits_chunk.float(), dim=-1)
+
+        ce_loss = 0.0
+        if compute_ce_loss:
+            # The hard/task loss
+            ce_loss = F.cross_entropy(
+                student_log_probs_chunk.view(-1, student_log_probs_chunk.shape[-1]),
+                target_chunk.view(-1),
+                reduction="sum",
+                ignore_index=ignore_index,
+            )
+
+        loss_mask = target_chunk != ignore_index
+        label_chunk = torch.where(loss_mask, target_chunk, 0)
+
+        student_average_log_prob = torch.zeros_like(loss_mask, dtype=torch.float)
+        student_per_token_logps = student_log_probs_chunk.gather(
+            -1, label_chunk.unsqueeze(-1)
+        ).squeeze(-1)
+
+        loss_mask_sum = loss_mask.sum(-1)
+        valid_mask = loss_mask_sum > 0
+
+        if valid_mask.any():
+            student_average_log_prob[valid_mask] = (
+                student_per_token_logps * loss_mask
+            ).sum(-1)[valid_mask] / loss_mask_sum[valid_mask]
+
+        # Teacher
+        teacher_logits_chunk = teacher_input_chunk @ teacher_weight.t()
+        if teacher_bias is not None:
+            teacher_logits_chunk = teacher_logits_chunk + teacher_bias
+        teacher_log_probs_chunk = F.log_softmax(teacher_logits_chunk.float(), dim=-1)
+
+        teacher_average_log_prob = torch.zeros_like(loss_mask, dtype=torch.float)
+        teacher_per_token_logps = teacher_log_probs_chunk.gather(
+            -1, label_chunk.unsqueeze(-1)
+        ).squeeze(-1)
+
+        if valid_mask.any():
+            teacher_average_log_prob[valid_mask] = (
+                teacher_per_token_logps * loss_mask
+            ).sum(-1)[valid_mask] / loss_mask_sum[valid_mask]
+
+        return student_average_log_prob, teacher_average_log_prob, ce_loss
+
+    @staticmethod
+    def forward(
+        ctx,
+        student_input,
+        student_weight,
+        teacher_input,
+        teacher_weight,
+        target,
+        student_bias=None,
+        teacher_bias=None,
+        loss_fn=None,
+        chunk_size=1,
+        ignore_index=-100,
+        beta=0.5,
```

Comment:
Perhaps we need another variable name for the weight between soft and hard loss, since some loss functions have 'beta' parameter, such as [generalized jsd](https://arxiv.org/abs/2306.13649) we've implemented in #278.